File: spring-data-requery/src/test/java/org/springframework/data/requery/repository/sample/UserRepositoryImpl.java
package org.springframework.data.requery.repository.sample;

import lombok.extern.slf4j.Slf4j;
import org.springframework.data.requery.domain.sample.User;

/**
 * UserRepositoryImpl
 *
 * @author debop@coupang.com
 * @since 18. 6. 25
 */
@Slf4j
public class UserRepositoryImpl implements UserRepositoryCustom {

    @Override
    public void findByOverrridingMethod() {
        log.debug("A method overriding a finder was invoked!");
    }

    @Override
    public void someCustomMethod(User user) {
        log.debug("Some custom method was invoked! user={}", user);
    }
}

File: spring-data-requery/src/test/kotlin/org/springframework/data/requery/repository/sample/UserRepositoryImpl.kt
package org.springframework.data.requery.repository.sample

import mu.KotlinLogging
import org.springframework.data.requery.domain.sample.User

/**
 * org.springframework.data.requery.repository.sample.UserRepositoryImpl
 *
 * @author debop
 */
class UserRepositoryImpl: UserRepositoryCustom {

    companion object {
        private val log = KotlinLogging.logger { }
    }

    override fun findByOverrridingMethod() {
        log.debug { "A method overriding a finder was invoked!" }
    }

    override fun someCustomMethod(user: User) {
        log.debug { "Some custom method was invoked! user=$user" }
    }
}